src/modules/schnorrsig/tests_exhaustive_impl.h

```diff
@@ -0,0 +1,211 @@
+        ((EXHAUSTIVE_TEST_ORDER + 0UL) >> 16) & 0xFF,
+        ((EXHAUSTIVE_TEST_ORDER + 0UL) >> 8) & 0xFF,
+        (EXHAUSTIVE_TEST_ORDER + 0UL) & 0xFF
+    },
+    /* order + 1 */
+    {
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        ((EXHAUSTIVE_TEST_ORDER + 1UL) >> 24) & 0xFF,
+        ((EXHAUSTIVE_TEST_ORDER + 1UL) >> 16) & 0xFF,
+        ((EXHAUSTIVE_TEST_ORDER + 1UL) >> 8) & 0xFF,
+        (EXHAUSTIVE_TEST_ORDER + 1UL) & 0xFF
+    },
+    /* field size */
+    {
+        0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+        0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFE, 0xFF, 0xFF, 0xFC, 0x2F
+    },
+    /* field size + 1 (note that 1 is legal) */
+    {
+        0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+        0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFE, 0xFF, 0xFF, 0xFC, 0x30
+    },
+    /* 2^256 - 1 */
+    {
+        0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+        0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF
+    }
+};
+
+#define NUM_INVALID_KEYS (sizeof(invalid_pubkey_bytes) / sizeof(invalid_pubkey_bytes[0]))
+
+static int secp256k1_hardened_nonce_function_smallint(unsigned char *nonce32, const unsigned char *msg32,
+                                                      const unsigned char *key32, const unsigned char *xonly_pk32,
+                                                      const unsigned char *algo16, void* data) {
+    secp256k1_scalar s;
+    int *idata = data;
+    (void)msg32;
+    (void)key32;
+    (void)xonly_pk32;
+    (void)algo16;
+    secp256k1_scalar_set_int(&s, *idata);
+    secp256k1_scalar_get_b32(nonce32, &s);
+    return 1;
+}
+
+static void test_exhaustive_schnorrsig_verify(const secp256k1_context *ctx, const secp256k1_xonly_pubkey* pubkeys, unsigned char (*xonly_pubkey_bytes)[32], const int* parities) {
+    int d;
+    uint64_t iter = 0;
+    /* Iterate over the possible public keys to verify against (through their corresponding DL d). */
+    for (d = 1; d <= EXHAUSTIVE_TEST_ORDER / 2; ++d) {
+        int actual_d;
+        unsigned k;
+        unsigned char pk32[32];
+        memcpy(pk32, xonly_pubkey_bytes[d - 1], 32);
+        actual_d = parities[d - 1] ? EXHAUSTIVE_TEST_ORDER - d : d;
+        /* Iterate over the possible valid first 32 bytes in the signature, through their corresponding DL k.
+           Values above EXHAUSTIVE_TEST_ORDER/2 refer to the entries in invalid_pubkey_bytes. */
+        for (k = 1; k <= EXHAUSTIVE_TEST_ORDER / 2 + NUM_INVALID_KEYS; ++k) {
+            unsigned char sig64[64];
+            int actual_k = -1;
+            int e_done[EXHAUSTIVE_TEST_ORDER] = {0};
+            int e_count_done = 0;
+            if (skip_section(iter++)) continue;
```

Comment:
`skip_section` takes a pointer, same in schnorrsig_sign

Reply:
Fixed.